src/Application.vala

```diff
@@ -137,45 +139,48 @@ public class Mail.Application : Gtk.Application {
     }
 
     public override void activate () {
-        if (run_in_background) {
-            run_in_background = false;
-            new InboxMonitor ().start.begin ();
-            hold ();
-            return;
-        }
-
-        MainWindow? main_window = null;
-        foreach (unowned var window in get_windows ()) {
-            if (window is MainWindow) {
-                main_window = (MainWindow) window;
-                break;
+        if (!run_in_background) {
+            MainWindow? main_window = null;
+            foreach (unowned var window in get_windows ()) {
+                if (window is MainWindow) {
+                    main_window = (MainWindow) window;
+                    break;
+                }
             }
-        }
 
-        if (main_window == null) {
-            main_window = new MainWindow (this);
-            add_window (main_window);
+            if (main_window == null) {
+                main_window = new MainWindow (this);
+                add_window (main_window);
 
-            int window_x, window_y;
-            var rect = Gtk.Allocation ();
+                int window_x, window_y;
+                var rect = Gtk.Allocation ();
 
-            settings.get ("window-position", "(ii)", out window_x, out window_y);
-            settings.get ("window-size", "(ii)", out rect.width, out rect.height);
+                settings.get ("window-position", "(ii)", out window_x, out window_y);
+                settings.get ("window-size", "(ii)", out rect.width, out rect.height);
 
-            if (window_x != -1 || window_y != -1) {
-                main_window.move (window_x, window_y);
-            }
+                if (window_x != -1 || window_y != -1) {
+                    main_window.move (window_x, window_y);
+                }
 
-            main_window.set_allocation (rect);
+                main_window.set_allocation (rect);
 
-            if (settings.get_boolean ("window-maximized")) {
-                main_window.maximize ();
+                if (settings.get_boolean ("window-maximized")) {
+                    main_window.maximize ();
+                }
+
+                main_window.show_all ();
             }
 
-            main_window.show_all ();
+            main_window.present ();
+        } else {
+            run_in_background = false;
         }
 
-        main_window.present ();
+        if (first_run) {
+            first_run = false;
```

Comment:
Would it be better to initialize the inbox monitor in `startup` instead of `activate`? I’m not sure about this `first_run` variable since it seems what we really care about is if the inbox monitor is running and the fact that it is started on first run is kind of incidental and we have a function for doing things once when the application is initialized

Reply:
The thing is in the future we want to ask the flatpak portal for autstart and background permission everytime the app is started and to make the dialog transient we need the MainWindow already active, so we will probably need this `first_run` (maybe it should be called something else more similar to startup) thingy anyway. Or maybe there is a better way to do that? Is it possible to call activate in startup without any complications?

Reply:
Another thing I forgot to mention was that we have to make sure that the `start ()` method from `Session` is called from the `MainWindow` first if mail is run without the background flag otherwise there might be some issues with the `WelcomeView` being briefly shown. An alternative here would be to move the entire `is_session_started`, `session_started` etc stuff into `Session` itself.